zerver/management/commands/send_zulip_updates.py

```diff
@@ -0,0 +1,134 @@
+from datetime import timedelta
+from typing import Any
+
+from django.conf import settings
+from django.db import transaction
+from django.db.models import Q
+from django.utils.timezone import now as timezone_now
+from typing_extensions import override
+
+from zerver.actions.message_send import (
+    do_send_messages,
+    internal_prep_private_message,
+    internal_prep_stream_message,
+)
+from zerver.lib.management import ZulipBaseCommand
+from zerver.models.realm_audit_logs import RealmAuditLog
+from zerver.models.realms import Realm
+from zerver.models.users import get_system_bot
+
+zulip_updates_file_path = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), "zulip_updates.md"
+)
+
+
+def get_latest_zulip_updates_level() -> int:
+    with open(zulip_updates_file_path) as zulip_updates:
+        content = zulip_updates.read()
+        match = re.search(r"\*\*Zulip updates level (\d+)\*\*", content)
+        assert match is not None
+        return int(match.group(1))
+
+
+def get_zulip_update_message_for_level(level: int) -> str:
+    with open(zulip_updates_file_path) as zulip_updates:
+        content = zulip_updates.read()
+        # TODO: Add a comment explaining algo in plain words.
+        current_level = f"**Zulip updates level {level}**"
+        previous_level = f"**Zulip updates level {level-1}**"
+        pattern = re.compile(
+            f"{re.escape(current_level)}(.*?)" + f"{re.escape(previous_level)}", re.DOTALL
+        )
+        match = re.search(pattern, content)
+        assert match is not None
+        return match.group(1).strip()
+
+
+class Command(ZulipBaseCommand):
+    help = """Script to send zulip updates to realms."""
+
+    @override
+    def handle(self, *args: Any, **options: str) -> None:
+        latest_zulip_updates_level = get_latest_zulip_updates_level()
+
+        realms = Realm.objects.filter(
+            Q(zulip_updates_level__isnull=True)
+            | Q(zulip_updates_level__lt=latest_zulip_updates_level),
+            deactivated=False,
+        )
+
+        for realm in realms:
+            realm_zulip_updates_level = realm.zulip_updates_level
+            sender = get_system_bot(settings.NOTIFICATION_BOT, realm.id)
+
+            messages = []
+            new_zulip_updates_level = None
+            with transaction.atomic():
+                if realm_zulip_updates_level is None:
+                    # realm predates the zulip updates feature.
+                    # DM the administrator to set or verify the stream for zulip updates.
+                    administrators = realm.get_human_admin_users()
+                    content = get_zulip_update_message_for_level(level=1)
+                    for administrator in administrators:
+                        direct_message = internal_prep_private_message(
+                            sender=sender, recipient_user=administrator, content=content
+                        )
+                        messages.append(direct_message)
```

Comment:
I think the idea we'd had in mind was a group DM to all of the administrators, so that they know exactly who received it and can easily coordinate by replacing there. Worth testing that replying to such a thread including Notification Bot works nicely, but I think it should -- and then we'd only have a single message ID that we can put in the audit log entry.

Reply:
Makes sense. Changed it to group DM.

I have used the term `group_direct_message` instead of `huddle`, (because, I think we're renaming huddle to group_direct_message -- I think there was a discussion on CZO ?) -- If I'm not correct, I will change it back to huddle.

Reply:
Yeah that sounds reasonable.